ios/keyman/Keyman/Keyman/Classes/KMWebBrowser/WebBrowserViewController.swift

```diff
@@ -285,24 +290,94 @@ class WebBrowserViewController: UIViewController, UIWebViewDelegate, UIAlertView
   }
 
   private func appendCSSFontFamily() {
-    let jsStr: String = "var style = document.createElement('style');" +
-      "style.type = 'text/css';" +
-      "style.innerHTML = '*{font-family:\"\(fontFamily)\" !important;}';" +
-    "document.getElementsByTagName('head')[0].appendChild(style);"
-    webView?.stringByEvaluatingJavaScript(from: jsStr)
+    guard let font = font else {
+      return
+    }
+
+    guard let fontKeyboard = fontKeyboard else {
+      return
+    }
+
+    let fontCSS = "*{font-family:\"\(font.family)\" !important;}"
+    guard let fontFaceStyle = buildFontSheet(from: font, keyboard: fontKeyboard) else {
+      return
+    }
+
+    let jsStr: String = """
+    var style = document.createElement('style');
+    style.type = 'text/css';
+    style.innerHTML = `
+    \(fontCSS)
+
+    \(fontFaceStyle)
+    `;
+
+    document.getElementsByTagName('head')[0].appendChild(style);
+    """
+
+    webView.evaluateJavaScript(jsStr)
+
+  }
+
+  private func buildFontSheet(from font: Font, keyboard: InstallableKeyboard) -> String? {
+    guard let fontKeyboard = fontKeyboard,
+          let fontURL = Manager.shared.fontPathForKeyboard(withFullID: fontKeyboard.fullID) else {
+      return nil
+    }
+
+    guard let fontData = NSData(contentsOf: fontURL) else {
+      return nil
+    }
+
+    var dataType: String = ""
+    var fontFormat: String = ""
+    switch fontURL.pathExtension {
+      case "ttf":
+        dataType = "font/truetype"
+        fontFormat = "truetype"
+      case "woff":
+        dataType = "font/woff"
+        fontFormat = "woff"
+      case "woff2":
+        dataType = "font/woff2"
+        fontFormat = "woff2"
+      case "otf":
+        dataType = "font/opentype"
+        fontFormat = "opentype"
+      case "eot":
+        fontFormat = "application/vnd.ms-fontobject"
+        fontFormat = "embedded-opentype"
+      case "svg":
+        dataType = "image/svg+xml"
+        fontFormat = "svg"
+      default:
+        return nil
+    }
+
+    let styleString = """
+    @font-face {
+      font-family: "\(font.family)";
+      src: url(data:\(dataType);charset=utf-8;base64,\(fontData.base64EncodedString())); format('\(fontFormat)')
+    }
+    """
+
+    return styleString
   }
 
   private func keyboardChanged(_ kb: InstallableKeyboard) {
-    if let fontName = Manager.shared.fontNameForKeyboard(withFullID: kb.fullID) {
-      newFontFamily = fontName
+    if let font = kb.font {
+      newFont = font
+      newFontKeyboard = kb
     } else {
-      newFontFamily = UIFont.systemFont(ofSize: UIFont.systemFontSize).fontName
+      newFont = nil
+      newFontKeyboard = nil
     }
   }
 
   private func keyboardPickerDismissed() {
-    if newFontFamily != fontFamily {
-      fontFamily = newFontFamily
+    if newFont?.family != font?.family {
+      font = newFont
+      fontKeyboard = newFontKeyboard
```

Comment:
I would like to point out that this is _exactly_ what the browser did before this PR's changes... so what you're asking me to do is to roll another (albeit, previously-undocumented) issue into this PR?

I feel like that may be best left as a separate, follow-up issue, since this one's already been through a few rounds and has a fair few things going on.

Reply:
After a bit more thought... I guess the check then boils down to:

1. If the previous keyboard had a `.font`:  "Did the user select the same keyboard?"
2. Otherwise:  "Did the user select a new keyboard that does have a set font?"